codesign/lib/src/file_codesign_visitor.dart

```diff
@@ -0,0 +1,129 @@
+abstract class FileVisitor {
+  const FileVisitor();
+
+  // TODO(xilaizhang): add back the visit interfaces
+}
+
+enum NotaryStatus {
+  pending,
+  failed,
+  succeeded,
+}
+
+/// Codesign and notarize all files within a [RemoteArchive].
+class FileCodesignVisitor extends FileVisitor {
+  FileCodesignVisitor({
+    required this.commitHash,
+    required this.codesignCertName,
+    required this.codesignUserName,
+    required this.appSpecificPassword,
+    required this.codesignAppstoreId,
+    required this.codesignTeamId,
+    required this.codesignFilepaths,
+    this.production = false,
+  });
+
+  /// Temp [Directory] to download/extract files to.
+  ///
+  /// This file will be deleted if [validateAll] completes successfully.
+  Directory? tempDir;
+  FileSystem? fileSystem;
+  Stdio? stdio;
+  ProcessManager? processManager;
+
+  final String commitHash;
+  final String codesignCertName;
+  final String codesignUserName;
+  final String appSpecificPassword;
+  final String codesignAppstoreId;
+  final String codesignTeamId;
+  final bool production;
+
+  // TODO(xilaizhang): add back utitlity in later splits
+  Set<String> fileWithEntitlements = <String>{};
+  Set<String> fileWithoutEntitlements = <String>{};
+  Set<String> fileConsumed = <String>{};
+  List<String> codesignFilepaths;
+
+  late final File entitlementsFile;
+  late final Directory remoteDownloadsDir;
+  late final Directory codesignedZipsDir;
+
+  int _remoteDownloadIndex = 0;
+  int get remoteDownloadIndex => _remoteDownloadIndex++;
+
+  static const String _entitlementsFileContents = '''
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+    <dict>
+        <key>com.apple.security.cs.allow-jit</key>
+        <true/>
+        <key>com.apple.security.cs.allow-unsigned-executable-memory</key>
+        <true/>
+        <key>com.apple.security.cs.allow-dyld-environment-variables</key>
+        <true/>
+        <key>com.apple.security.network.client</key>
+        <true/>
+        <key>com.apple.security.network.server</key>
+        <true/>
+        <key>com.apple.security.cs.disable-library-validation</key>
+        <true/>
+    </dict>
+</plist>
+''';
+
+  void initialize() {
+    fileSystem ??= LocalFileSystem();
+    tempDir ??= fileSystem!.systemTempDirectory.createTempSync('conductor_codesign');
+    stdio ??= VerboseStdio(
+      stdout: io.stdout,
+      stderr: io.stderr,
+      stdin: io.stdin,
+    );
+    processManager ??= LocalProcessManager();
+    entitlementsFile = tempDir!.childFile('Entitlements.plist')..writeAsStringSync(_entitlementsFileContents);
+    remoteDownloadsDir = tempDir!.childDirectory('downloads')..createSync();
+    codesignedZipsDir = tempDir!.childDirectory('codesigned_zips')..createSync();
+  }
+
+  Future<void> validateAll() async {
+    initialize();
+
+    try {
```

Comment:
Remove the unnecessary try/catch?

Reply:
Yeah I am putting a try catch block here because the `validateAll` function (the entrance point) is going to trigger other functions which are programmed to throw many kinds of exceptions. Should I remove the try catch block in this PR and add it back later?

Reply:
Yes, let's remove and add when its needed

Reply:
I would try to avoid this pattern of expecting to catch all exceptions. We do this in the conductor because it is intended as an end user product (run by non-engineers), so we handle exceptions with pretty user messages.

Since this is an automated script intended to be run on CI, I think it's better to either:

1. if an exception is expected and recoverable, handle it closer to the call site; or
2. if it is unrecoverable, let it crash the script

Reply:
Thanks  a lot for the input! Removed.